pineappl/src/grid.rs

```diff
@@ -758,6 +804,93 @@ impl Grid {
         Ok(())
     }
 
+    /// Return a vector containing the type of convolutions performed with this grid.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the metadata key--value pairs `convolution_particle_1` and `convolution_type_1`,
+    /// or `convolution_particle_2` and `convolution_type_2` are not correctly set.
+    #[must_use]
+    pub fn convolutions(&self) -> Vec<Convolution> {
```

Comment:
although I suspect the function does a fairly simple job, it looks complicated - could you please add some more (code) comments or break the function into smaller pieces?

Reply:
I added more comments in commit ba94ab9cf1a300b7727c57ed4d05a87b86788f93 - this function will become a simple getter once we save the convolution types as members of a `Grid` using a new file format.